lib/internment.rs

```diff
@@ -0,0 +1,233 @@
+use differential_datalog::record;
+use differential_datalog::record::*;
+use internment::ArcIntern;
+use serde;
+use std::cmp;
+use std::fmt;
+
+#[cfg(feature = "flatbuf")]
+use flatbuf::{FromFlatBuffer, ToFlatBuffer, ToFlatBufferTable, ToFlatBufferVectorElement};
+
+/* `flatc`-generated declarations re-exported by `flatbuf.rs` */
+#[cfg(feature = "flatbuf")]
+use flatbuf::fb;
+
+/* FlatBuffers runtime */
+#[cfg(feature = "flatbuf")]
+use flatbuffers as fbrt;
+
+#[derive(Default, Eq, PartialOrd, PartialEq, Ord, Clone, Hash)]
+pub struct internment_Intern<A>
+where
+    A: Eq + Send + Hash + 'static,
+{
+    intern: ArcIntern<A>,
+}
+
+impl<A: Eq + Hash + Send + 'static> internment_Intern<A> {
+    pub fn new(x: A) -> internment_Intern<A> {
+        internment_Intern {
+            intern: ArcIntern::new(x),
+        }
+    }
+    pub fn as_ref(&self) -> &A {
+        self.intern.as_ref()
+    }
+}
+
+pub fn internment_intern<A: Eq + Hash + Send + Clone + 'static>(x: &A) -> internment_Intern<A> {
+    internment_Intern::new(x.clone())
+}
+
+pub fn internment_ival<A: Eq + Hash + Send + Clone>(x: &internment_Intern<A>) -> A {
+    x.intern.as_ref().clone()
+}
+
+/*pub fn intern_istring_ord(s: &intern_IString) -> u32 {
+    s.x
+}*/
+
+impl<A: fmt::Display + Eq + Hash + Send + Clone> fmt::Display for internment_Intern<A> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(self.as_ref(), f)
+        //record::format_ddlog_str(&intern_istring_str(self), f)
```

Comment:
Because this is very likely to cause problems for the new implementation and I will need to figure out how to fix it. At least there will be a reminder in the code.